app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,64 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, only: [:edit, :update] 
+  before_action :valid_user, only: [:edit, :update] 
+  before_action :check_expiration, only: [:edit, :update] # Case (1)
+ 
+  def new 
+  end
+  def create
+    @user = User.find_by(email: params[:password_reset][:email].downcase) 
```

Comment:
```suggestion
    @user = User.find_by(email: email_param) 
```
always best to optimize for readability by extracting non-essential details into methods/variables